hal/src/boron/network/sara_ncp_client.cpp

```diff
@@ -498,53 +498,66 @@ int SaraNcpClient::getSignalQuality(CellularSignalQuality* qual) {
     CHECK(checkParser());
     CHECK(queryAndParseAtCops(qual));
 
+    // Min and max RSRQ index values multiplied by 100
+    // Min: -19.5 and max: -3
+    const int min_rsrq_mul_by_100 = -1950;
+    const int max_rsrq_mul_by_100 = -300;
+
     if (ncpId() == PLATFORM_NCP_SARA_R410) {
-        int rxlev, rxqual, rscp, ecn0, rsrq, rsrp;
-        auto resp = parser_.sendCommand("AT+CESQ");
-        int r = CHECK_PARSER(resp.scanf("+CESQ: %d,%d,%d,%d,%d,%d", &rxlev, &rxqual,
-                &rscp, &ecn0, &rsrq, &rsrp));
-        CHECK_TRUE(r == 6, SYSTEM_ERROR_AT_RESPONSE_UNEXPECTED);
-        r = CHECK_PARSER(resp.readResult());
-        CHECK_TRUE(r == AtResponse::OK, SYSTEM_ERROR_AT_NOT_OK);
+        int rsrp;
+        int rsrq_n;
+        unsigned long rsrq_f;
+
+        // Default to 255 in case RSRP/Q are not found
+        qual->strength(255);
+        qual->quality(255);
+
+        // Set UCGED to mode 5 for RSRP/RSRQ values on R410M
+        CHECK_PARSER_OK(parser_.execCommand("AT+UCGED=5"));
+        auto resp = parser_.sendCommand("AT+UCGED?");
+        int r1 = 0;
+        int r2 = 0;
+        while (resp.hasNextLine()) {
+            r1 = CHECK_PARSER(resp.scanf("+RSRP: %*d,%*d,\"%d.%*u\"",&rsrp));
+            r2 = CHECK_PARSER(resp.scanf("+RSRQ: %*d,%*d,\"%d.%lu\"",&rsrq_n, &rsrq_f));
+        }
 
-        switch (qual->strengthUnits()) {
```

Comment:
I'd propose we leave something similar here to check that RAT acquired in `queryAndParseAtCops()` is in fact LTE.

Reply:
I removed that because R410M supports only LTE. TMK either it acquires LTE or it doesn't. Am I missing something?

Reply:
I can leave the switch case as-in and put in this additional stuff.

Reply:
The upper layer code expects RAT to be available. So maybe it's fine to leave as-is. Let's leave as-is.

Reply:
OK.. 
You mean I need to add the switch case or should we not add the switch case and leave it as-it-is?
The one thing I am OCD about is if ublox supports multiple RATs on this R410M, or if we start supporting NB-IoT on this that will be a different RAT. If the upper layer is still taking care of this, it's probably OK to not add the switch case?

Reply:
I think it is fine to leave things as-is. Even if RAT is not set properly by COPS, the whole thing will be ignore by upper layer code.

Reply:
For future reference - Even if it's NB-IoT, it will still be RSRP/Q params for signal strength